Add WKFrameInfo accessor to _WKJSHandle

achristensen07 opened the pull request:
#### fccd92e3da0195681cd6f07e33366e1716e95fe2
<pre>
Add WKFrameInfo accessor to _WKJSHandle
<a href="https://bugs.webkit.org/show_bug.cgi?id=297745">https://bugs.webkit.org/show_bug.cgi?id=297745</a>
<a href="https://rdar.apple.com/158885679">rdar://158885679</a>

Reviewed by Ryosuke Niwa.

A _WKJSHandle can&apos;t be used for much without using a WKFrameInfo, and it has to be the correct
WKFrameInfo, so before this change we always needed to keep the pair of objects together.
This makes the API a little more ergonomic by adding a frame accessor so you only need to keep
one object in a container and can use it.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::frameID const): Deleted.
* Source/WebCore/dom/Document.h:
(WebCore::Document::frameID const):
* Source/WebKit/Modules/OSX_Private.modulemap:
* Source/WebKit/Modules/iOS_Private.modulemap:
* Source/WebKit/Shared/API/c/WKBase.h:
* Source/WebKit/Shared/JSHandleInfo.cpp: Copied from Source/WebKit/Shared/JSHandleInfo.h.
* Source/WebKit/Shared/JSHandleInfo.h:
* Source/WebKit/Shared/JavaScriptEvaluationResult.cpp:
(WebKit::JavaScriptEvaluationResult::APIInserter::toAPI):
(WebKit::JavaScriptEvaluationResult::APIExtractor::toValue):
(WebKit::JavaScriptEvaluationResult::JSExtractor::toValue):
(WebKit::JavaScriptEvaluationResult::JSInserter::toJS):
* Source/WebKit/Shared/JavaScriptEvaluationResult.h:
* Source/WebKit/Shared/JavaScriptEvaluationResult.mm:
(WebKit::JavaScriptEvaluationResult::ObjCInserter::toID):
(WebKit::JavaScriptEvaluationResult::ObjCExtractor::toValue):
* Source/WebKit/Shared/JavaScriptEvaluationResult.serialization.in:
* Source/WebKit/Shared/NodeHitTestResult.h:
* Source/WebKit/Shared/NodeHitTestResult.serialization.in:
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/C/WKAPICast.h:
* Source/WebKit/UIProcess/API/C/WKJSHandleRef.cpp: Copied from Source/WebKit/Shared/JSHandleInfo.h.
(WKJSHandleGetTypeID):
(WKJSHandleCopyFrameInfo):
* Source/WebKit/UIProcess/API/C/WKJSHandleRef.h: Copied from Source/WebKit/Shared/JSHandleInfo.h.
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _hitTestAtPoint:inFrameCoordinateSpace:completionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKJSHandle.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKJSHandle.mm:
(-[_WKJSHandle frame]):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::hitTestAtPoint):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::hitTestAtPoint):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::(SiteIsolation, HitTesting)):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::Callbacks::append):
(WTR::TestController::Callbacks::notifyListeners):
(WTR::CompletionHandler&lt;void):
* Tools/WebKitTestRunner/TestController.h:
(WTR::TestController::Callbacks::append): Deleted.

Canonical link: <a href="https://commits.webkit.org/299058@main">https://commits.webkit.org/299058@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6126bb2ef71ff2bdd18bb2538db17981a40a831b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117588 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37265 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27890 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123700 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69596 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/8585fe96-6673-4b85-8a95-cf0d5cd26d4f) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119466 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37958 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45848 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89236 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/45046 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2ae142e2-7def-4836-a0de-2055c00262b0) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120540 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30212 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105429 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/69738 "Build was cancelled. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bf76b22b-7b4c-4bdb-aedc-08a2c80a35eb) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29274 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23545 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67369 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99628 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23727 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126813 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44491 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33471 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97902 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44849 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101658 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97690 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43062 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21014 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/40849 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18778 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44362 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50037 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43820 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47168 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45512 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0399e323860070ffd40cb5855ef255d9298b9a81. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49729

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fff09cc0cf6fe2761b3305ebbbc9d3f672202770. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49729

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/42f2c73c2601b818ec6124a7665a690bb660d204. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49729

achristensen07 committed:
Add WKFrameInfo accessor to _WKJSHandle
https://bugs.webkit.org/show_bug.cgi?id=297745
rdar://158885679

Reviewed by Ryosuke Niwa.

A _WKJSHandle can't be used for much without using a WKFrameInfo, and it has to be the correct
WKFrameInfo, so before this change we always needed to keep the pair of objects together.
This makes the API a little more ergonomic by adding a frame accessor so you only need to keep
one object in a container and can use it.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::frameID const): Deleted.
* Source/WebCore/dom/Document.h:
(WebCore::Document::frameID const):
* Source/WebKit/Modules/OSX_Private.modulemap:
* Source/WebKit/Modules/iOS_Private.modulemap:
* Source/WebKit/Shared/API/c/WKBase.h:
* Source/WebKit/Shared/JSHandleInfo.cpp: Copied from Source/WebKit/Shared/JSHandleInfo.h.
* Source/WebKit/Shared/JSHandleInfo.h:
* Source/WebKit/Shared/JavaScriptEvaluationResult.cpp:
(WebKit::JavaScriptEvaluationResult::APIInserter::toAPI):
(WebKit::JavaScriptEvaluationResult::APIExtractor::toValue):
(WebKit::JavaScriptEvaluationResult::JSExtractor::toValue):
(WebKit::JavaScriptEvaluationResult::JSInserter::toJS):
* Source/WebKit/Shared/JavaScriptEvaluationResult.h:
* Source/WebKit/Shared/JavaScriptEvaluationResult.mm:
(WebKit::JavaScriptEvaluationResult::ObjCInserter::toID):
(WebKit::JavaScriptEvaluationResult::ObjCExtractor::toValue):
* Source/WebKit/Shared/JavaScriptEvaluationResult.serialization.in:
* Source/WebKit/Shared/NodeHitTestResult.h:
* Source/WebKit/Shared/NodeHitTestResult.serialization.in:
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/C/WKAPICast.h:
* Source/WebKit/UIProcess/API/C/WKJSHandleRef.cpp: Copied from Source/WebKit/Shared/JSHandleInfo.h.
(WKJSHandleGetTypeID):
(WKJSHandleCopyFrameInfo):
* Source/WebKit/UIProcess/API/C/WKJSHandleRef.h: Copied from Source/WebKit/Shared/JSHandleInfo.h.
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _hitTestAtPoint:inFrameCoordinateSpace:completionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKJSHandle.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKJSHandle.mm:
(-[_WKJSHandle frame]):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::hitTestAtPoint):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::hitTestAtPoint):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::(SiteIsolation, HitTesting)):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::Callbacks::append):
(WTR::TestController::Callbacks::notifyListeners):
(WTR::CompletionHandler<void):
* Tools/WebKitTestRunner/TestController.h:
(WTR::TestController::Callbacks::append): Deleted.

Canonical link: https://commits.webkit.org/299058@main
